internal/dao/dao_errors.go

```diff
@@ -8,10 +8,12 @@ import (
 
 var (
 	// ErrNoRows is returned when there are no rows in the result
+	// Typically, REST requests should end up with 404 error
 	ErrNoRows = pgx.ErrNoRows
 
 	// ErrAffectedMismatch is returned when unexpected number of affected rows
 	// was returned for INSERT, UPDATE and DELETE queries.
+	// Typically, REST requests should end up with 404 error
 	ErrAffectedMismatch = errors.New("unexpected affected rows")
```

Comment:
Hmm when you update or delete something and it fails, it should not be 404 but perhaps something else. I suggest [409 Conflict](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409) perhaps.